src/core/mac/sub_mac.cpp

```diff
@@ -843,6 +843,11 @@ void SubMac::SetMacKey(uint8_t    aKeyIdMode,
     case Frame::kKeyIdMode2:
         break;
     case Frame::kKeyIdMode1:
+
+        IgnoreError(otPlatCryptoDestroyKey(mPrevKey.GetKeyRef()));
+        IgnoreError(otPlatCryptoDestroyKey(mCurrKey.GetKeyRef()));
+        IgnoreError(otPlatCryptoDestroyKey(mNextKey.GetKeyRef()));
```

Comment:
@abtink If KeyRefs are not used we `otPlatCryptoDestroyKey` would not be implemented and would be an empty function in crypto_platform.cpp, so it will be an empty call.

We need to destroy these keys here because after generating the keys, key_manager will not store these references, so when new keys are created, it does not have the old keys to destroy them, unless we instantiate Mac in key_manager, which I wanted to avoid.

Yes, I thought of using it as a list and destroying only the previous key, but this will not work if the NetworkKey itself is changed, There is not way for Mac to know if the update is because of networkkey update or keySeqnumber update. It is better to assume that these are new keys and treat them this way.

Reply:
I understand it's easier to call `otPlatCryptoDestroyKey()` here instead of in `KeyManager` (since it won't remember/save them). I think it's not the best model but It is fine and we can keep these calls here (I have been wanting to do an enhancement where we move all key and security related stuff that is now in `SubMac` into a module of its own, and we can address it when we add such enhancement)

Coupe of suggestions here:
- How about adding `#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE` around these calls? 
   -  This way we don't need to rely on the platform behavior that `otPlatCryptoDestroyKey` should do nothing if not enabled
   - It is a safer model and platform implementation may choose to add an implementation `otPlatCryptoDestroyKey()` without it being enabled/used immediately (platform implementer may not realize that they need to have empty implementation thinking it won't be used unless PSA is enabled).

- I think we use `keyRef` value of zero as unused. We init them all as zero (calling `Clear()` on `mPrevKey`, `mCurKey` , `mNextkey` from constructor).
- I think we either need to check in the `SubMac::SetMacKey()` and skip calling `otPlatCryptoDestroyKey()` if the key ref is zero, or document that  we expect `otPlatCryptoDestroyKey()` would do nothing if the key ref passed to it is zero.
   - It is always better to be very explicit and explain in detail behaviors and expectations of the platform APIs. A platform developer may not look into the OT core code to see how the APIs are being used and just read and rely on the API docs only.